### ISSUE TITLE
Incompatibility with Craft 3.1 beta

### ISSUE THREAD

**Issue by elivz** [verbb/feed-me#435]
### Description

Craft 3.1 switched from storing relationship field sources as integer IDs to UIDs. As a result, on the "map fields" step of setting up a new import, an error gets thrown, similar to the following:

    Argument 1 passed to craft\services\Volumes::getVolumeById() must be of the type integer, string given, called in /var/www/dist/vendor/verbb/feed-me/src/web/twig/variables/FeedMeVariable.php on line 122

### Steps to reproduce

1. Create an asset field (presumably you would get the same result with an entries field, category field, etc.)
2. Create a new import task and proceed to the "Map Fields" step.

### Additional info

- Plugin version: 3.0.0
- Craft version: 3.1.0-beta1

**Comment by McCrisp** [comment]
I am getting this same error on Craft 3.0.33

**Comment by elivz** [comment]
Pull request #438 appears to fix the problem!

**Comment by elivz** [comment]
Hold on. #438 fixes the error when mapping field, but if I actually try to import some entries that include Asset fields, I still get the `Argument 1 passed to craft\services\Assets::getFolderById() must be of the type integer, string given, called in /var/www/dist/vendor/smcyr/feed-me/src/fields/Assets.php on line 61 - Assets.php: 378` message in my logs and those items all fail to import.

**Comment by smcyr** [comment]
I updated #438 to fix all source ids, let me know if it works now. Thanks

**Comment by engram-design** [comment]
Yep, only just starting compatibility with 3.1 beta, as I didn't want to start too early when things were still broken or up in the air.

@smcyr Many thanks on your work for the PR and giving things a head start.

**Comment by jdsimcoe** [comment]
I am getting `Internal Server Error: Servor Error` messages on the latest `3.1.0-beta-5.1`:

![image](https://user-images.githubusercontent.com/275617/50249187-27d6ae00-0392-11e9-9dd9-c18174bfef8c.png)

I get this trying to move past the `Primary Element` phase.

**Comment by thisisjamessmith** [comment]
Thanks for the fix @smcyr - just a quick note that the field mapping screen still fails to load if you're importing categories to a category group that doesn't have any fields. I just worked around it by adding a custom field to the category group.

**Comment by engram-design** [comment]
@thisisjamessmith Fixed in the latest 3.1 branch

**Comment by engram-design** [comment]
@jdsimcoe Would you be able to provide the full error with devMode on?

**Comment by jdsimcoe** [comment]
@engram-design I will work on it.

**Comment by jdsimcoe** [comment]
@engram-design Here is my full error output: https://gist.github.com/jdsimcoe/d36487e3b3b0a37138e0c3080c4274de.

**Comment by mark-chief** [comment]
Hey @engram-design, Just wondering how you were getting on with this? I am running in to these problems the craft 3.1.1. if it's fixed in the other branch, how do i install that version via composer? Thanks!
